scene/resources/packed_scene.cpp

```diff
@@ -287,11 +288,20 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 									// tampering the resource object already there with the values from the node in the parent scene and
 									// then tell this node to reference that resource.
 									if (n.instance >= 0) {
-										Ref<Resource> node_res = node->get(snames[nprops[j].name]);
-										if (node_res.is_valid()) {
-											node_res->copy_from(res);
-											node_res->configure_for_local_scene(node, resources_local_to_scene);
-											value = node_res;
+										HashMap<Ref<Resource>, Ref<Resource>>::Iterator E = resources_local_to_sub_scene.find(res);
+										if (E) {
+											value = E->value;
+										} else {
+											Ref<Resource> node_res = node->get(snames[nprops[j].name]);
```

Comment:
You may want to add a bit of commenting here as well, for future maintainers to make sense of the code more easily, now details are still fresh in your mind.